ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -766,37 +827,60 @@ def runTest(self):
-
-            self.log("Wait until data plane stops")
-            async_forward_stop = pool.apply_async(self.check_forwarding_stop)
-
-            try:
-                async_cpu_up.get(timeout=self.task_timeout)
-            except TimeoutError as e:
-                self.log("DUT hasn't booted up in %d seconds" % self.task_timeout)
-                self.fails['dut'].add("DUT hasn't booted up in %d seconds" % self.task_timeout)
-                raise
-
-            try:
-                no_routing_start, upper_replies = async_forward_stop.get(timeout=self.task_timeout)
-                self.log("Data plane was stopped, Waiting until it's up. Stop time: %s" % str(no_routing_start))
-            except TimeoutError:
-                self.log("Data plane never stop")
-                no_routing_start = datetime.min
-
-            if no_routing_start is not None:
-                self.timeout(self.task_timeout, "DUT hasn't started to work for %d seconds" % self.task_timeout)
-                no_routing_stop, _ = self.check_forwarding_resume()
-                self.cancel_timeout()
-            else:
-                no_routing_stop = datetime.min
+            if self.reboot_type == 'fast-reboot':
+                self.log("Check that device is still forwarding data plane traffic")
+                self.fails['dut'].add('Data plane has a forwarding problem')
+                self.assertTrue(self.check_alive(), 'DUT is not stable')
+                self.fails['dut'].clear()
+
+                self.log("Wait until control plane up")
+                async_cpu_up = pool.apply_async(self.wait_until_cpu_port_up)
+
+                self.log("Wait until data plane stops")
+                async_forward_stop = pool.apply_async(self.check_forwarding_stop)
+
+                try:
+                    async_cpu_up.get(timeout=self.task_timeout)
+                except TimeoutError as e:
+                    self.log("DUT hasn't bootup in %d seconds" % self.task_timeout)
+                    self.fails['dut'].add("DUT hasn't booted up in %d seconds" % self.task_timeout)
+                    raise
+
+                try:
+                    no_routing_start, upper_replies = async_forward_stop.get(timeout=self.task_timeout)
+                    self.log("Data plane was stopped, Waiting until it's up. Stop time: %s" % str(no_routing_start))
+                except TimeoutError:
+                    self.log("Data plane never stop")
+                    no_routing_start = datetime.min
+
+                if no_routing_start is not None:
+                    self.timeout(self.task_timeout, "DUT hasn't started to work for %d seconds" % self.task_timeout)
+                    no_routing_stop, _ = self.check_forwarding_resume()
+                    self.cancel_timeout()
+                else:
+                    no_routing_stop = datetime.min
+
+                # Stop watching DUT
+                self.watching = False
+
```

Comment:
You can probably indent left above 2 lines and remove the 2 more copies of stop below before finally

Reply:
I removed it in the place right before the 'BGP check',
And left if at:
at the end of fast-reboot clause.
at the beginning of warm-reboot clause.
and at finally clause.